src/geopathfinder/naming_conventions/yeoda_naming.py

```diff
@@ -319,11 +303,10 @@ def yeoda_path(root, product=None, data_version=None, datalog='datasets', grid=N
         levels.remove('qlooks')
         hierarchy.remove('qlook')
 
-    return create_smartpath(root, hierarchy=hierarchy, levels=levels,
-                     make_dir=make_dir)
+    return create_smartpath(root, hierarchy=hierarchy, levels=levels, make_dir=make_dir)
 
 
-def yeoda_tree(root, target_level=None, register_file_pattern=None):
+def yeoda_tree(root, target_level=None, register_file_pattern=None, grid_pattern=('EQUI7')):
```

Comment:
In general I think it would make your life easier (and also that of users) if any pattern in `geopathfinder` is always a regex pattern, then the user can decide how strings should be filtered. But thats a different topic and must not be solved here.

Reply:
they are then treated as regex patterns.

As I need always to look them up, I prefer to just provide them as tuple.

--> should be improved all over the package

Reply:
Yes I know, and it works smoothly also by providing a more complicated regex pattern as one tuple entry, but you only concatenate them via `.*` and offer a negative look-ahead for a special character. I know, regex are awful to get along with, but I don't think one needs to reinvent the wheel here. However, what you can do is to provide this functionality in a separate keyword to make certain tasks easier for the user: `(should_contain=('Qlook', 'S1A'), should_not_contain=('test'))` this might be a bad example in terms of variable names, but then its in my opinion easier to understand and to steer from the outside what happens.

Reply:
sounds good. but right now this behaviour can be achieved with
grid_pattern=('Qlook', 'S1A', '-test')